city_scrapers/spiders/il_port_district.py

```diff
@@ -0,0 +1,181 @@
+
+                meeting = Meeting(
+                    title=title,
+                    description="",
+                    classification=classification,
+                    start=start,
+                    end=None,
+                    all_day=False,
+                    time_notes="",
+                    location=self.location,
+                    links=links,
+                    source=response.url
+                )
+
+                meeting["status"] = self._get_status(meeting)
+                meeting["id"] = self._get_id(meeting)
+
+                yield meeting
+
+    def parse_agendas(self, response):
+        file_prefix = response.xpath("//tr/td[@class='views-field views-field-title']/text()")\
+            .extract()
+        file_prefix = [x.strip("\n ") for x in file_prefix]
+        file_links = response.xpath("//tr/td/a[@class='file-download']/@href").extract()
+        name_pattern = r"(?<=documents\/)(.*?)(?=.pdf)"
+        file_names = [re.findall(name_pattern, x)[0].replace('%', ' ') for x in file_links]
+
+        self.files_list = []
+
+        for link, prefix, name in list(zip(file_links, file_prefix, file_names)):
+            self.files_list.append({'title': prefix + " " + name, 'href': link})
+
+    def parse_minutes(self, response):
+        rows = response.xpath("//tr")
+        self.minutes_dict = {}
+        for row in rows:
+            file_name = row.xpath(".//td[@class='views-field views-field-title']/text()")\
+                .extract_first()
+            if not file_name:
+                continue
+
+            file_name = file_name.strip("\n ")
+            file_name_dt = re.findall(r"(?:.*?)(?:\d{4})", file_name)[0]
+            file_name_dt = datetime.datetime.strptime(file_name_dt, "%B %d, %Y")
+
+            file_link = row.xpath(".//td[@class='views-field views-field-field-file']/a/@href")\
+                .extract_first()
+
+            self.minutes_dict.setdefault(file_name_dt, file_link)
+
+    def _parse_classification(self, i, meeting_types):
+        if "board" in meeting_types.lower():
+            return BOARD
+        elif "committee" in meeting_types.lower():
+            return COMMITTEE
+        else:
+            return NOT_CLASSIFIED
+
+    def _parse_start(self, year, date, meeting_time):
+        dt = " ".join([year, date, meeting_time])
+        return datetime.datetime.strptime(dt, "%Y %B %d %I:%M%p")
+
+    def _parse_agendas_links(self, classification):
+        if classification == 'Board':
+            link_list = [d for d in self.files_list if 'Board' in d['title']]
+        elif classification == 'Committee':
+            link_list = [d for d in self.files_list if 'Committee' in d['title']]
+        else:
+            link_list = [{}]
+
+        return link_list
+
+    def _parse_minutes_links(self, start):
+        start = start.replace(hour=0, minute=0, second=0, microsecond=0)
+        link = self.minutes_dict.get(start, None)
```

Comment:
Instead of replacing the details, it's been easier to use a `defaultdict` object keyed by `date` rather than `datetime` objects so that you can call `minutes_dict[start.date()]`